R/bootstrap.R

```diff
@@ -1492,78 +1492,79 @@ downloadLink <- function(outputId, label="Download", class=NULL, ...) {
-#'   [Glyphicons](https://getbootstrap.com/components/#glyphicons)
-#'   libraries. Note that the "fa-" and "glyphicon-" prefixes should not be used
-#'   in icon names (i.e. the "fa-calendar" icon should be referred to as
-#'   "calendar")
-#' @param class Additional classes to customize the style of the icon (see the
+#' of a button, or as an icon for a [tabPanel()] within a [navbarPage()].
+#'
+#' @param name The name of the icon. Icons are drawn from the [Font Awesome
+#'   Free](https://fontawesome.com/) (currently icons from the v5.13.0 set are
+#'   supported with the v4 naming convention) and
+#'   [Glyphicons](https://getbootstrap.com/docs/3.3/components/#glyphicons)
+#'   libraries. Note that the `"fa-"` and `"glyphicon-"` prefixes should not be
+#'   used in icon names (i.e. the `"fa-calendar"` icon should be referred to as
+#'   `"calendar"`).
+#' @param class Additional classes to customize the style of an icon (see the
 #'   [usage examples](https://fontawesome.com/how-to-use) for details on
 #'   supported styles).
-#' @param lib Icon library to use ("font-awesome" or "glyphicon")
-#' @param ... Arguments passed to the `<i>` tag of [htmltools::tags]
+#' @param lib The icon library to use. If drawing from the Font Awesome library
+#'   of icons, this should be `"font-awesome"`. The other option is or
+#'   `"glyphicon"` to use icons from the Glyphicons library.
+#' @param ... Arguments passed to the `<i>` tag of [htmltools::tags].
 #'
-#' @return An icon element
+#' @return An icon element.
 #'
 #' @seealso For lists of available icons, see
 #'   [https://fontawesome.com/icons](https://fontawesome.com/icons) and
-#'   [https://getbootstrap.com/components/#glyphicons](https://getbootstrap.com/components/#glyphicons).
-#'
+#'   [https://getbootstrap.com/docs/3.3/components/#glyphicons](https://getbootstrap.com/docs/3.3/components/#glyphicons).
 #'
 #' @examples
 #' # add an icon to a submit button
-#' submitButton("Update View", icon = icon("refresh"))
+#' submitButton("Update View", icon = icon("redo"))
 #'
 #' navbarPage("App Title",
 #'   tabPanel("Plot", icon = icon("bar-chart-o")),
 #'   tabPanel("Summary", icon = icon("list-alt")),
 #'   tabPanel("Table", icon = icon("table"))
 #' )
 #' @export
-icon <- function(name, class = NULL, lib = "font-awesome", ...) {
+icon <- function(name,
+                 class = NULL,
+                 lib = "font-awesome",
+                 ...) {
+
   prefixes <- list(
     "font-awesome" = "fa",
     "glyphicon" = "glyphicon"
   )
   prefix <- prefixes[[lib]]
 
-  # determine stylesheet
+  # Stop function if `lib` value is not allowed
   if (is.null(prefix)) {
-    stop("Unknown font library '", lib, "' specified. Must be one of ",
-         paste0('"', names(prefixes), '"', collapse = ", "))
+    stop(
+      "Unknown font library '", lib, "' specified. Must be one of ",
+      paste0('"', names(prefixes), '"', collapse = ", "),
+      call. = FALSE
```

Comment:
I think this `call.=FALSE` will make the message more confusing, since users won't see where the error is coming from.

Reply:
I agree and it's gone now.